homeassistant/components/fully_kiosk/sensor.py

```diff
@@ -51,6 +51,7 @@ class FullySensorEntityDescription(SensorEntityDescription):
 SENSORS: tuple[FullySensorEntityDescription, ...] = (
     FullySensorEntityDescription(
         key="batteryLevel",
+        translation_key="battery_level",
         device_class=SensorDeviceClass.BATTERY,
```

Comment:
These sensors have automatic translation, this should not be needed. Something is setting a name elsewhere.

Reply:
I.e any sensor with device class has translation by default. So you need to figure out what breaks that.

Reply:
Thanks, @elupus. Digging into this I see that to derive the name `Entity` depends on `EntityPlatform.component_translations`. However, this is always an empty `dict` for the sensor domain, which ends up causing `Entity.name` to return `None`. The only way I can get the name is by setting the `translation_key` or `name` parameters when instantiating `FullySensorEntityDescription`.

Reply:
There are tests for the battery sensor, which would fail if the name wasn't correct:

https://github.com/home-assistant/core/blob/d6a0f9b5a004d8b8ca865204db1d7b37b6d219fe/tests/components/fully_kiosk/test_sensor.py#L37-L46

Have you generated the translations for the sensor integration when experiencing the issue?

Reply:
Yes, the test not passing is what got me into this. But it is resolved by generating the translations for the sensor integration, as you pointed out. I thought that translations for these base integrations where automatically generated. By the way, is there a way to generate the translations for all the integrations?

Reply:
The translations are automatically generated for the production build, not when running a local development environment. Then you need to explicitly generate the translations for the integration you're working on.

Reply:
I always run the script and it asks which integration to generate for. I was wondering if it can generate for all integrations, to spare myself problems like this down the road.

Reply:
Yes there is. Check the help option output of the script.

Reply:
I had looked at that but `python -m script.translations develop -h` doesn't say. However, looking at develop.py, it accepts the --all flag, which does the trick.